[Samuel Yip] iP

samyipsh opened the pull request:
## DukeV2
> Cos who doesn't love having their own quote :scroll:
> -Samuel

DukeV2 allows you to store different types of task seamlessly in your local storage 
- Text-based
- **Free** to use
- Easy to use

#### Running the program
1. Download jar file from [latest release version](https://github.com/samyipsh/ip/releases/tag/v0.1)
2. Enter cmd prompt > enter command `java -jar <pathToFile>`
3. Alternatively
```
cd <directoryPath>
java -jar <fileName.jar>
```

### Extension roll out
- [x] v0.1
- [ ] v0.2
- [ ] v0.3
- [ ] 
 Tasks type | Parameters
------------ | -------------
ToDo | taskSummary
Deadline | taskSummary, time (dd/MM/uuuu HHmm)
Event | taskSummary, time (string)

Commit:
Add Gradle support

samyipsh committed:
add level-1 code

samyipsh committed:
add Level-2 files, to do list

samyipsh committed:
add Level-3 files, feature to toggle task done

samyipsh committed:
begin Level-4, add ToDo.java class

samyipsh committed:
add Level-4, classes of ToDo, Deadline, Event

samyipsh committed:
complete level-4, add Duke, Event class

samyipsh committed:
add A-TextUiTesting, running on windows with .bat file

samyipsh committed:
handle errors

samyipsh committed:
Add new exception class DukeException, aka add A-Exceptions tag

samyipsh committed:
add Level-6 Delete functionality, changed ToDoList task field from Task[] to List<Task>, updated testing

samyipsh committed:
update testing

samyipsh committed:
add Storage class and write method

samyipsh committed:
refactor code - change type to final TYPE

samyipsh committed:
add storage.write(String) functionality

samyipsh committed:
add storage.write(ArrayList<Task> tasks) functionality
add Level-7 Save & Load

samyipsh committed:
add functionality in Deadline to parse and format string to LocalDateTime (MAY BREAK)

samyipsh committed:
Merge branch 'branch-Level-7'

samyipsh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java

samyipsh committed:
handle errors for deadline command. finish level-7 level-8

samyipsh committed:
modify regex of deadline, event, todo

samyipsh committed:
add Ui class (more oop)

samyipsh committed:
add Parser class, reformat Duke

samyipsh committed:
add packages and reformat code with intellij suggestions

samyipsh committed:
add JUnit 5.4 support (TextUItest file path now wrong)

samyipsh committed:
add equals method for ToDo, Event, Deadline tasks

samyipsh committed:
add JUnit test for equals method for Event, ToDo, Deadline

samyipsh committed:
add  fat .jar file, build artifiact with dependencies

samyipsh committed:
add javadoc to classes

samyipsh committed:
modify code to be closer to coding standard

samyipsh committed:
add find command.

samyipsh committed:
Merge branch 'branch-Level-9'

samyipsh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/util/Parser.java

tanruiquan commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
I think you made the same mistake as me by using the refactor button in IntelliJ which also changed name in the README file, you might want to consider changing it back.

tanruiquan commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,131 @@
+        DateTimeFormatter.ofPattern("dd-MM-uuuu HHmm");
+
+    /**
+     * Factor method of Deadline class
```
Do you mean `Factory`?

tanruiquan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public Duke(String filePathToStorage) {
+        Storage DUKE_STORE = new Storage(filePathToStorage);
+        TaskList TASKLIST = TaskList.of(DUKE_STORE);
```
Since `DUKE_STORE` and `TASKLIST` is not final, maybe you can consider camelCase instead? 
```suggestion
        Storage dukeStore = new Storage(filePathToStorage);
        TaskList taskList = TaskList.of(DUKE_STORE);
```

tanruiquan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+    public Duke(String filePathToStorage) {
+        Storage DUKE_STORE = new Storage(filePathToStorage);
+        TaskList TASKLIST = TaskList.of(DUKE_STORE);
+        this.PARSER = Parser.initialize(TASKLIST);
+        this.UI = new Ui();
```
Should `this.PARSER` and `this.UI` simply be `PARSER` and `UI` as they are not shadowed by a method or constructor parameters? I noticed this issue in several other places too, maybe it is something for you to consider?
```suggestion
        PARSER = Parser.initialize(TASKLIST);
        UI = new Ui();
```

samyipsh replied:
haha yea I was thinking about changing it, cos in cs2030 they said put this.x for like everything so kinda stuck to it, will change it tho thx

tanruiquan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.Arrays;
+import java.util.List;
+
+//make abstract class?
```
You might want to avoid writing comments as if they are private notes to yourself as stated in the SE textbook [here](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#write-to-the-reader)

tanruiquan reviewed (commented):
Overall, the code is really well written and i can see that you put in a substantial amount of effort to adhere to Software Engineering principles. Just a few nits to fix and it LGTM.

samyipsh committed:
1. Commented out buggy code (checking if date has passed in Deadline)
2. Changed var names to better meet coding standard

samyipsh committed:
Merge branch 'add-gradle-support' of https://github.com/samyipsh/ip into branch-a-gradle

samyipsh committed:
set up gradle (merged gradle branches from upstream repo)

samyipsh committed:
[**pending error fix**] import checkstyle files from AB3

samyipsh committed:
add some setup for javaFx in `build.gradle` and `Duke.java`

samyipsh committed:
fix checkstyle problems: add @skyblaise99 checkstyle file and it works

samyipsh committed:
disable checkstyle options in `build.gradle` **temporarily** to work on other tasks

samyipsh committed:
add FXML support files (**able to build**) (**BUT unable to run Main.java**)

samyipsh committed:
add .fxml integration
- **CANT RUN** - `Caused by: java.lang.IllegalStateException: Location is not set.`
- **can build**

samyipsh committed:
fix bug by adding in missing 's' in MainWindows.fxml reference

samyipsh committed:
integrate logic of replying user input to GUI MainWindows. Modify Duke.java and Ui.java

tchiong commented on src/main/java/duke/controller/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
+    }
```
Perhaps you could add docs to clarify these 2 statics methods are they could potentially be combined into a single method with a boolean to determine if a flip is needed

tchiong commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,133 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
```
Could perhaps add a header docs to this class

tchiong reviewed (commented):
LGTM! Other than the comments, the code is readable and clear

samyipsh committed:
modify css and fxml files to improve looks

samyipsh committed:
Refactor code:
Code has some unexplained regex and in-line variables

Abstracted out such to relevant classes like TaskList to
avoid confusion.

TaskList is currently abstracting out syntax of commands. Fits the
abstraction it has.

To do
*Extract out more unexplained regex from Parser class

samyipsh committed:
minor tweaks to Parser to make it more readable

samyipsh committed:
Refactor of code & Addition of asserts

* Shifted logic of execution of task from Parser to TaskList
* Added asserts to some functions in TaskList (toggleDoneAndUpdateStore
& removeTaskAndUpdateStore

samyipsh committed:
Merge branch 'branch-add-assertions'

samyipsh committed:
Replace TaskList's functions with Streams
* a more functional approach
* 2 functions changed: toString() & selectTasksAsString

samyipsh committed:
Merge pull request #1 from samyipsh/branch-add-streams

Replace TaskList's functions with Streams

samyipsh committed:
setup github actions with given yml file

samyipsh committed:
Merge pull request #2 from samyipsh/branch-github-ci

setup github actions with given yml file

samyipsh committed:
* add help functionality
* refactor code to put **user-facing code** into Ui class

samyipsh committed:
add isExistingUser private variable into Storage for future use.

samyipsh committed:
Merge branch 'branch-add-help'

samyipsh committed:
set stage to be not resizable

samyipsh committed:
modify main windows gui
* expand mainwindow and components by 50px
* change font in DialogBox to monospace 'Courier New'

samyipsh committed:
improve gui
* modify Text back to Label
* add background fill and radius curve

samyipsh committed:
add circle clipping to imageView

samyipsh committed:
Merge branch 'branch-better-gui'

samyipsh committed:
clean up code
* remove commented out obsolete code
* add some javadoc

samyipsh committed:
add user guide (docs/README.md)

samyipsh committed:
change duke name to meap (Change of project application name)

samyipsh committed:
create jar file for release 0.2
